### PR TITLE
Make _seen_jti a ringbuffer.

### PR DESCRIPTION
**Pull request by erikvanzijst**
Throwing away the entire contents of the jti list when it reaches capacity seems undesirable. This change makes it a ringbuffer that automatically takes care of expunging old entries.

I've also increased the capacity.

### CODE REVIEW

File: atlassian_jwt_auth/verifier.py [hunk 1]
```diff
@@ -11,7 +13,7 @@ class JWTAuthVerifier(object):
     def __init__(self, public_key_retriever, **kwargs):
         self.public_key_retriever = public_key_retriever
         self.algorithms = algorithms.get_permitted_algorithm_names()
-        self._seen_jti = set()
+        self._seen_jti = deque(maxlen=1000)
```
Comment (dbaxa) [inline atlassian_jwt_auth/verifier.py]: I don't mind having a backing deque like object but the (average) time complexity is for searching for a jti is higher for `deque` than it is for a `set` (O(n) vs O(1)).

Comment (erikvanzijst) [inline atlassian_jwt_auth/verifier.py]: That's fair. I'll switch it to log(n).

File: atlassian_jwt_auth/verifier.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+from collections import deque, OrderedDict
```
Comment (dbaxa) [inline atlassian_jwt_auth/verifier.py]: We don't need the `deque` import any longer.

**Comment by dbaxa** [comment]
Will re-raise.

**Pull request closed without merging**